homeassistant/components/deconz/scene.py

```diff
@@ -31,11 +32,14 @@ def async_add_scene(
         | ValuesView[PydeconzScene] = gateway.api.scenes.values(),
     ) -> None:
         """Add scene from deCONZ."""
-        entities = [
-            DeconzScene(scene, gateway)
-            for scene in scenes
-            if scene.deconz_id not in gateway.entities[DOMAIN]
-        ]
+        entities = []
+
+        for scene in scenes:
+
+            known_entities = set(gateway.entities[DOMAIN])
+            new_entity = DeconzScene(scene, gateway)
+            if new_entity.unique_id not in known_entities:
```

Comment:
```
binary_sensor.py:            known_entities = set(gateway.entities[DOMAIN])
binary_sensor.py:                if new_sensor.unique_id not in known_entities:
number.py:            known_entities = set(gateway.entities[DOMAIN])
number.py:                if new_entity.unique_id not in known_entities:
sensor.py:            known_entities = set(gateway.entities[DOMAIN])
sensor.py:                if new_entity.unique_id not in known_entities:
```

It looks like this pattern is repeated in other places.  Is there an opportunity to make this a bit more DRY?

Reply:
Yeah, first step towards that is breaking out how to generate serial number and unique ID